### PR TITLE
Some love to the logger

### PR DESCRIPTION
**Pull request by alfuken**
Can't say if it works because I can't run it, but maybe it's just me. It's 3 AM here, so I'll live it at that. Please have a look and maybe it'll compile and run on your end. Anyway, it should show the idea about general direction...

UPD. After using an `-all.jar`, it seems to be working. But needs testing.

### CODE REVIEW

File: src/main/java/com/superzanti/serversync/config/SyncConfig.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
     public List<String> FILE_IGNORE_LIST = Arrays.asList("**/serversync-*.jar", "**/serversync-*.cfg");
     public List<String> CONFIG_INCLUDE_LIST = new ArrayList<>();
     public Locale LOCALE = Locale.getDefault();
-    public ETheme THEME = ETheme.BLUE_YELLOW;
```
Comment (rheimus) [inline src/main/java/com/superzanti/serversync/config/SyncConfig.java:28]: Looks like an unrelated change to logging. Should probably not change this anyway as it is the UI author's original intent and user-configurable.

File: src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ public void start(Stage primaryStage) throws Exception {
         Scene scene = new Scene(root, 1024, 576);
         scene.getStylesheets().add(this.getClass().getResource("/css/application.css").toExternalForm());
         primaryStage.setScene(scene);
+        primaryStage.sizeToScene();
```
Comment (rheimus) [inline src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java:33]: Unrelated to logging?

File: src/main/java/com/superzanti/serversync/GUIJavaFX/PaneOptions.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import com.superzanti.serversync.ServerSync;
 import com.superzanti.serversync.config.SyncConfig;
-import com.superzanti.serversync.util.Logger;
+import com.superzanti.serversync.util.ServerSyncLogger;
```
Comment (rheimus) [inline src/main/java/com/superzanti/serversync/GUIJavaFX/PaneOptions.java:5]: Considering that nothing about the callsite of this class appears to have changed it would have been much easier to review the changeset if the name of the class remained unchanged or was changed as part of a different commit.

File: src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java (hunk 2 above)
Comment (alfuken) [reply src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java:33]: Yep, unrelated. This was done with version 3 in mind and an assumption that UI hasn't changed, as on Macs it was pretty darn tiny, like 300px wide or something. UI in v4 is much bigger now, thanks. But is it resizeable, though?..

File: src/main/resources/css/application.css [hunk 4]
```diff
@@ -27,8 +27,8 @@
     -fx-text-fill: -fx-primarytext;
     -fx-background-color: -fx-primary;
     -fx-border-color: -fx-secondary;
-    -fx-border-radius: 20px;
```
Comment (rheimus) [inline src/main/resources/css/application.css:30]: Unrelated to logging?

File: src/main/java/com/superzanti/serversync/util/ServerSyncLogger.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import com.superzanti.serversync.ServerSync;
+import com.superzanti.serversync.files.PathBuilder;
+import javafx.application.Platform;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.logging.*;
```
Comment (rheimus) [inline src/main/java/com/superzanti/serversync/util/ServerSyncLogger.java]: Nice, makes sense to use built in java features 👍

File: src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java (hunk 2 above)
Comment (rheimus) [reply src/main/java/com/superzanti/serversync/GUIJavaFX/Gui_JavaFX.java]: I believe it is user resizable, we can revisit the default size in a different pull request though if you would like.

**Comment by rheimus** [comment]
Looks like the log file for individual connections has resources being held open after the client disconnects.

Probably just needs to be destroyed after the client exits.

**Comment by rheimus** [comment]
We probably need to change the logger to be non static as well, the current version will not allow the instance specific logging required for the "per connection" logging.
